File: Dataset/Leetcode/train/53/485.cpp
 class Solution {
public:
    int XXX(vector<int>& nums) {
        int max = nums[0];
        for(int i = 1; i < nums.size();i++){
            if(nums[i] + nums[i-1] > nums[i]){
                nums[i] = nums[i] + nums[i-1];
            }
            if(nums[i] > max){
                max = nums[i];
            }
        }
        return max;
    }
};

File: Dataset/Leetcode/test/110/485.java
class Solution {
    public boolean XXX(TreeNode root) {
        return digui(root)>=0;
    }

    public int digui(TreeNode root) {
        if(root==null){
            return 0;
        }
        int left = digui(root.left);
        int right = digui(root.right);
        if(left<0 || right<0 || left-right>1 || right-left>1){
            return -1;
        }
        return Math.max(left,right)+1;
    }
}

